### ISSUE TITLE
Can't run VPN Manager from 'My Addons' in Add-On Browser 

### ISSUE THREAD

**Issue by Raubion** [Zomboided/service.vpn.manager#363]
I recently switched from Libreelec to OSMC. It has a slightly different UI, [about which I posted on their forums](https://discourse.osmc.tv/t/add-on-category-overview-as-main-menu-item-not-my-addons/92619/16). One of the conclusions of the thread is that VPN Manager is one of few addons that can't be run from the 'My Addons' section in the Add-On Browser. It can only be configured. Other addons, can also be run from there. 

Is there a reason for this? And would it be possible to make it launchable from this section?

**Comment by Zomboided** [comment]
Hello, I’ve read the thread but I’m not clear what the issue is. VPN Mgr is following standard Kodi design patterns for displaying lists (at least it was when I wrote it and still seemed to be when it got updated for Python 3 last year).

I’ve used this across Windows and LibreELEC with multiple different skins with no issue.  I’ve not used OSMC and have not plans to try it, but the issue seems to be with a particular skin.

If there are minor changes I can make to allow it to work, then I need to know what they are (and they need not to break compatibility with other skins).  I don’t have the energy to work out what’s wrong in your specific use case myself tho, sorry.

**Comment by peno64** [comment]
Why? Once configured you dont need to be in there anymore.

**Comment by peno64** [comment]
"I would want to access this to check if VPN is connected or to disconnect or reconnect."
You can define shortrcut keys for that. I use F1, F2 for this. I can  be anywhere in kodi, when I press F2 I get  connection info. If I pess F1 I also get  a notification  and if I long press F1 I can connect to another region or disconnect .

**Comment by Raubion** [comment]
Right, that presumes use of keyboard though. I use my remote mostly or otherwise Yatse. I guess I could also create shortcuts for that somewhere too. But I'd like to stress it works just fine with the program add-ons item added to the main menu. 

As I said, I'm mostly just curious why other add-ons can be launched/ran/started from the My Addons menu, but VPN Manager can't.

**Comment by Zomboided** [comment]
I use a remote control with a hot key to check/cycle the connection, but
usually I just trust that it's set up correctly and connects and
disconnects when required.  You can remote control or hot keys up using
built in functionality to let you set this up without having to edit
keymaps.

Back on topic, I can see this behaviour in the Estuary skin too now you've
provided the extra info.  Each add-on will describe the catagory it's in at
a high level and the entry points that it provides. i.e.
https://kodi.wiki/view/Addon.xml (section 1.4).  VPN Mgr is correctly
defined as a xbmc.service (as it runs all the time) and a
xbmc.python.pluginsource to enable the menu access.  One of my other
add-ons that does video things, is correctly defined as a
xbmc.addon.video.  The video add-on has run enabled.  I've also looked at
other services, such as a the version checker which also does not offer the
run option.

I'm not a skin developer, but from my understanding, it's up to the skin
developer to choose what views offer what actions with what add-ons - i.e.
on a video add-on view, you'd not display services add-ons.  I'm guessing
that in the My Addons view of Estuary, the skin author has chosen to not
offer the run option with services add-ons.  This could be entirely
reasonable as not all services would also have additional functionality
through a menu.

Anyway, my conclusion here is that there's nothing my add-on is doing
that's incorrect and I've no desire to change it to pretend to be a video
add-on for instance as that would cause all sorts of other undesireable
issues.  Maybe you can convince skin authors to change their functionality,
but I'm not sure it'd be appropriate for all service add-ons.  You've been
given workarounds for this, I suggest you pursue those.
